src/amoc_throttle/amoc_throttle.erl

```diff
@@ -1,6 +1,148 @@
 %%==============================================================================
-%% Copyright 2023 Erlang Solutions Ltd.
-%% Licensed under the Apache License, Version 2.0 (see LICENSE file)
+%% @doc This module allows to synchronize the users and act on groups of them.
```

Comment:
Being honest, I'm not a big fun of such big documentation blocks in the header. I would rather keep a separated markdown doc (just maybe move it into the `guides` directory, so all the documentation is at one place).